src/modules/form/components/OccurrencePointForm.tsx

```diff
@@ -1,75 +1,61 @@
+import IconButtonContainer from '@/modules/enrollment/components/IconButtonContainer';
 import DynamicView from '@/modules/form/components/viewable/DynamicView';
+import { SubmitFormInputVariables } from '@/modules/form/hooks/useDynamicFormHandlersForRecord';
 import { useFormDialog } from '@/modules/form/hooks/useFormDialog';
-import { FormValues, isQuestionItem } from '@/modules/form/types';
+import {
+  LocalConstants,
+  PickListArgs,
+  SubmitFormAllowedTypes,
+} from '@/modules/form/types';
 import {
   AlwaysPresentLocalConstants,
   createInitialValuesFromRecord,
-  getItemMap,
-  getInitialValues,
-  modifyFormDefinition,
   getDisabledLinkIds,
+  getInitialValues,
+  getItemMap,
 } from '@/modules/form/util/formUtil';
-import { DashboardEnrollment } from '@/modules/hmis/types';
 import {
   FormDefinitionFieldsFragment,
   FormDefinitionJson,
-  FormItem,
   FormRole,
   InitialBehavior,
 } from '@/types/gqlTypes';
 
-function matchesTitle(item: FormItem, title: string) {
-  return !![item.text, item.readonlyText].find(
-    (s) => s && s.toLowerCase() === title.toLowerCase()
-  );
-}
-
-export const parseOccurrencePointFormDefinition = (
-  definition: FormDefinitionFieldsFragment
-) => {
-  let displayTitle = definition.title;
-  let isEditable = false;
-  const readOnlyDefinition = modifyFormDefinition(
-    definition.definition,
-    (item) => {
-      if (definition.title && matchesTitle(item, definition.title)) {
-        displayTitle = item.readonlyText || item.text || displayTitle;
-        delete item.text;
-        delete item.readonlyText;
-      }
-      if (isQuestionItem(item) && !item.readOnly) {
-        isEditable = true;
-      }
-    }
-  );
-
-  return { displayTitle, isEditable, readOnlyDefinition };
-};
-
-function hasAnyValues(object: FormValues) {
-  return !!Object.keys(object).find((k) => !isNil(object[k]));
-}
-
-interface OccurrencePointValueProps {
-  enrollment: DashboardEnrollment;
+export interface OccurrencePointFormProps {
+  record: SubmitFormAllowedTypes;
   definition: FormDefinitionFieldsFragment;
+  submitFormInputVariables?: SubmitFormInputVariables;
   readOnlyDefinition: FormDefinitionJson;
   editable?: boolean;
   dialogTitle?: string;
+  localConstants?: LocalConstants;
+  pickListArgs?: PickListArgs;
 }
 
-const OccurrencePointValue: React.FC<OccurrencePointValueProps> = ({
-  enrollment,
+/**
+ * This component renders the values from the `record` into the `definition` and displays
+ * it as a read-only view. If any of the `definition` items are not read-only, it shows
+ * a pencil icon button. When clicked, it brings up a modal that renders a form according to
+ * the `definition`, and submits changes using SubmitForm.
+ *
+ * This is currently primarily used for viewing/editing Custom Data Elements that are associated with
+ * a Client or an Enrollment, on the Client Dashboard and Enrollment Dashboard respectively.
+ */
+const OccurrencePointForm: React.FC<OccurrencePointFormProps> = ({
```

Comment:
This component probably has a poor name, but I couldn't come up with a good name for the component. I stuck with occurrence point form since thats currently _how_ we're using it.

Reply:
I realize that probably means this component is doing too much..